src/main/java/org/zalando/nakadi/service/subscription/zk/AbstractZkSubscriptionClient.java

```diff
@@ -382,15 +381,15 @@ public final void resetCursors(final List<SubscriptionCursorWithoutToken> cursor
     }
 
     @Override
-    public final Optional<ZkSubscriptionNode> getZkSubscriptionNodeLocked()
+    public final Optional<ZkSubscriptionNode> getZkSubscriptionNode()
             throws SubscriptionNotInitializedException, NakadiRuntimeException {
         if (!isSubscriptionCreatedAndInitialized()) {
             return Optional.empty();
         }
 
-        return Optional.of(runLocked(() -> new ZkSubscriptionNode(
```

Comment:
Why have you removed running under lock?

Reply:
Because it isn't necessary anymore, since we migrated to new zk subscription format (I double checked with @antban )